aggregation-worker/src/main/java/au/org/emii/geoserver/client/HttpIndexReader.java

```diff
@@ -0,0 +1,219 @@
+package au.org.emii.geoserver.client;
+
+
+import java.io.BufferedInputStream;
+import java.io.DataInputStream;
+import java.io.InputStream;
+import java.net.HttpURLConnection;
+import java.net.URI;
+import java.net.URL;
+import java.net.URLEncoder;
+import java.util.*;
+
+import au.org.emii.aggregator.exception.AggregationException;
+import au.org.emii.download.DownloadRequest;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+
+public class HttpIndexReader implements IndexReader {
+    private static final Logger logger = LoggerFactory.getLogger(HttpIndexReader.class);
+
+    protected String geoserver = null;
+
+    public HttpIndexReader(String geoserver) {
+        this.geoserver = geoserver;
+    }
+
+    @Override
+    public URIList getUriList(String profile, String timeField, String urlField, SubsetParameters subset) throws AggregationException {
+        String timeCoverageStart = subset.get("TIME").start;
+        String timeCoverageEnd = subset.get("TIME").end;
+
+        URIList uriList = new URIList();
+
+        try
+        {
+            String downloadUrl = String.format("%s/wfs", geoserver);
+            String cqlFilter = String.format("%s >= %s AND %s <= %s",
```

Comment:
Need to handle optional time range at some point

Reply:
As in cater for the case where no time params are passed?

Reply:
Yes - time is an optional subset parameter - some datasets don't include time and a time range wont be passed for them.